controller/backing_image_controller.go

```diff
@@ -287,30 +283,27 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
 	return nil
 }
 
-func (bic *BackingImageController) IsBackingImageDataSourceCleanup(bi *longhorn.BackingImage) (finished bool, err error) {
+func (bic *BackingImageController) IsBackingImageDataSourceCleaned(bi *longhorn.BackingImage) (cleaned bool, err error) {
 	bids, err := bic.ds.GetBackingImageDataSource(bi.Name)
-	if err != nil && !apierrors.IsNotFound(err) {
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// can not get backing image data source, return cleaned so the controller won't wait until bids become failed-and-cleanup
+			return true, nil
+		}
 		return false, err
 	}
 
+	// if bids is transferred, there is no tmp file left
 	if bids.Spec.FileTransferred {
 		return true, nil
 	}
 
-	// when bids status is ready or ready for transfer, there is already no tmp file left
-	if bids.Status.CurrentState == longhorn.BackingImageStateFailedAndCleanUp ||
-		bids.Status.CurrentState == longhorn.BackingImageStateReady ||
-		bids.Status.CurrentState == longhorn.BackingImageStateReadyForTransfer {
+	// if bids is not transferred, wait until it is failed-and-cleanup
+	if bids.Status.CurrentState == longhorn.BackingImageStateFailedAndCleanUp {
 		return true, nil
 	}
 
-	// mark the status to failed so manager can clean up the orphan tmp file
-	bids.Status.Message = "backing image is deleted, try to cleaning up the tmp file of data source"
-	bids.Status.CurrentState = longhorn.BackingImageStateFailed
-	if _, err = bic.ds.UpdateBackingImageDataSourceStatus(bids); err != nil {
-		return false, err
-	}
-	return false, nil
+	return false, fmt.Errorf("backing image data source status is %v not failed-and-cleanup", bids.Status.CurrentState)
```

Comment:
Replace `failed-and-cleanup` in log with `longhorn.BackingImageStateFailedAndCleanUp`